sql/src/main/java/com/vmware/ddlog/translator/TranslationVisitor.java

```diff
@@ -1040,11 +1040,50 @@ private DDlogIRNode processWindows(QuerySpecification query,
         ExpressionTreeRewriter<Void> subst = new ExpressionTreeRewriter<Void>(rewriter);
         ExpressionTreeRewriter<Void> dropTable = new ExpressionTreeRewriter<Void>(new ColumnContextEliminationRewriter());
 
-        for (SingleColumn sc: Utilities.concatenate(aggregateItems, nonAggregateItems, windowItems)) {
-            Expression repl = subst.rewrite(sc.getExpression(), null);
-            Expression repl1 = dropTable.rewrite(repl, null);
-            finalItems.add(new SingleColumn(repl1, sc.getAlias()));
+        // Add columns to finalItems in the order that they appear in the original table.
+        for (SelectItem originalColumn : query.getSelect().getSelectItems()) {
+            if (originalColumn instanceof AllColumns) {
+                Table r = (Table) query.getFrom().get();
+                String rn = DDlogRelationDeclaration.relationName(r.getName().toString());
+                DDlogType tableType = context.resolveType(context.getRelation(rn)
+                        .getType());
+                DDlogTStruct struct = tableType.to(DDlogTStruct.class);
+                for (DDlogField f: struct.getFields()) {
+                    Identifier id = new Identifier(f.getName());
+                    finalItems.add(new SingleColumn(id, id));
+                }
+            } else if (originalColumn instanceof SingleColumn) {
+                SingleColumn scOrigCol = (SingleColumn) originalColumn;
+                boolean found = false;
+                for (SingleColumn sc: Utilities.concatenate(aggregateItems, windowItems)) {
+                    // Aggregate or window items should be referenced by their name in the OverInput or Over tables
+                    // created by this Visitor.
+                    if (scOrigCol.getExpression().equals(sc.getExpression())) {
+                        Expression repl = subst.rewrite(sc.getExpression(), null);
+                        Expression repl1 = dropTable.rewrite(repl, null);
```

Comment:
maybe we should rename the 'dropTable' variable as 'dropTableSpecRewriter' or something like that.

Reply:
How about `dropTableExprRewriter`?